pkg/odo/cli/component/create.go

```diff
@@ -329,6 +326,14 @@ func (co *CreateOptions) Complete(name string, cmd *cobra.Command, args []string
 			co.CommonPushOptions.componentContext = co.componentContext
 		}
 
+		co.Context = genericclioptions.NewDevfileContext(cmd)
+
+		// One exception of a validation present in Complete code because, this is an optimisation
+		// i.e. if the devfile is present locally then we dont need to list the devfile catalog
+		if util.CheckPathExists(DevfilePath) {
```

Comment:
I don't think we can do the global check for devfile existence here, and we have to check separately in interactive mode and direct mode, reason is:
- Interactive mode: we can use devfile directly if devfile is present rather than error out, use can still specify the component name via interactive mode
- Direct mode: we need to error out as component type that user specify may have conflict with the existing devfile

Reply:
done